File: LeetCode/cpp/1071.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        int m = str1.size(), n = str2.size();
        if (m < n)
            return gcdOfStrings(str2, str1);
        for (int i = 1; i <= n; ++i) {
            if (n % i != 0 || m % (n / i) != 0)
                continue;
            string gcd = str2.substr(0, n / i);
            int s = gcd.size();
            bool flag = true;
            for (int j = 0; j < m / (n / i); ++j) {
                if (str1.substr(j * s, s) != gcd) {
                    flag = false;
                    break;
                }
            }
            if (flag)
                return gcd;
        }
        return "";
    }
};

File: LeetCode/python/1071.py
class Solution(object):
    def gcdOfStrings(self, str1, str2):
        """
        :type str1: str
        :type str2: str
        :rtype: str
        """
        m, n = len(str1), len(str2)
        if m < n:
            return self.gcdOfStrings(str2, str1)
        for i in range(1, n + 1):
            if n % i != 0 or m % (n // i) != 0:
                continue
            gcd = str2[:n // i]
            s = len(gcd)
            flag = True
            for j in range(m // (n // i)):
                if str1[j * s: j * s + s] != gcd:
                    flag = False
                    break
            if flag:
                return gcd
        return ""
